crates/transpiler/src/passes/consolidate_blocks.rs

```diff
@@ -44,6 +52,88 @@ pub enum DecomposerType {
     TwoQubitControlledU(TwoQubitControlledUDecomposer),
 }
 
+fn get_matrix(gate: &StandardGate) -> ArrayView2<Complex64> {
+    match gate {
+        StandardGate::CX => aview2(&CX_GATE),
+        StandardGate::CY => aview2(&CY_GATE),
+        StandardGate::CZ => aview2(&CZ_GATE),
+        StandardGate::CH => aview2(&CH_GATE),
+        StandardGate::DCX => aview2(&DCX_GATE),
+        StandardGate::ISwap => aview2(&ISWAP_GATE),
+        StandardGate::ECR => aview2(&ECR_GATE),
+        _ => unreachable!("Unsupported gate"),
+    }
+}
+
+/// Helper function that extracts the decomposer and basis gate directly from the [Target].
+#[inline]
+fn get_decomposer_and_basis_gate(
+    target: Option<&Target>,
+    approximation_degree: f64,
+) -> (DecomposerType, StandardGate) {
+    if let Some(target) = target {
+        // Targets from C should only support Standard gates.
+        if let Some(gate) = target.operations().find_map(|op| {
+            op.operation
+                .try_standard_gate()
+                .and_then(|gate| PARAM_SET_BASIS_GATES.contains(&gate).then_some(gate))
```

Comment:
If you actually did:
```suggestion
                .and_then(|gate| matches!(gate, PARAM_SET!())).then_some(gate))
```
and pulling the macro in form unitary synthesis. I think it would be potentially slightly more efficient, but it's really a small micro optimization.

Reply:
Fixed in cb637cc3f74f3788ad8b8108c0f13bc222c87c35